This patch was written by Janne Kukonlehto  (jtklehto@stekt.oulu.fi),
it adds 8 bit clean support to ncurses 1.8.5

--- ncurses-1.8.5/src/lib_addch.c.old	Sun Oct 30 21:03:08 1994
+++ ncurses-1.8.5/src/lib_addch.c	Sun Oct 30 21:04:59 1994
@@ -32,6 +32,13 @@
 	if (y > win->_maxy  ||  x > win->_maxx  ||  y < 0  ||  x < 0)
 	    return(ERR);
 
+	/* Attempt to solve problems caused by
+	   sign-extension of eighth bit 
+	   (this should make ncurses 8-bit clean) */
+	if ((ch & A_ATTRIBUTES) == A_ATTRIBUTES){
+		ch &= A_CHARTEXT;
+	}
+
 	switch (ch&A_CHARTEXT) {
     	case '\t':
 		for (newx = x + (8 - (x & 07)); x < newx; x++)
